### PR TITLE
fix: update linters config

### PR DESCRIPTION
**Pull request by JulioCVaz**
## Infos

[Task](https://juntossomosmais.monday.com/boards/950575490/pulses/5485719057)

#### What is being delivered?

Adjust linters package eslint config to new approach configuration

#### What impacts?

- Adjust eslint to valid configuration with numbers instead string("on"|"off")
- Add eslint-import-resolver-node to fix esmodules require

#### Reversal plan

Revert to stable version

#### Evidences

https://github.com/juntossomosmais/time-out-market/assets/20933173/8d4d934e-182f-47e8-8272-b59957fd4c1d

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "eslint": "^8.49.0",
     "eslint-import-resolver-typescript": "^3.6.1",
     "eslint-plugin-import": "^2.29.0",
+    "eslint-import-resolver-node": "^0.3.9",
```
Comment (felipefialho) [inline package.json:39]: (question) we need it inside the internal projects too? it yes we need to add as peer dependencies inside `packages/linters/package.json` and update instructions on `packages/linters/README.md`

File: packages/linters/src/eslint.config.react.js [hunk 2]
```diff
@@ -10,12 +10,12 @@ module.exports = {
...
-    'react/no-unescaped-entities': 'off',
-    'react/jsx-uses-react': 'on',
-    'react/react-in-jsx-scope': 'on',
-    'react/jsx-boolean-value': ['error', 'always'],
+    'react/display-name': 0,
+    'react/prop-types': 0,
+    'react/no-unescaped-entities': 0,
+    'react/jsx-uses-react': 1,
+    'react/react-in-jsx-scope': 1,
+    'react/jsx-boolean-value': [2, 'always'],
```
Comment (felipefialho) [inline packages/linters/src/eslint.config.react.js:18]: (question) the rules about `packages/linters/src/eslint.config.js` still working?

File: package.json (hunk 1 above)
Comment (JulioCVaz) [reply package.json:39]: @felipefialho Even with the dependency as peerdependency I had a problem committing the file https://github.com/juntossomosmais/time-out-market/blob/main/packages/linters/src/eslint.config.react.js#L1. Basically was this error https://stackoverflow.com/questions/44865507/eslint-no-duplicates-resolve-error-unable-to-load-resolver-node.

File: packages/linters/src/eslint.config.react.js (hunk 2 above)
Comment (JulioCVaz) [reply packages/linters/src/eslint.config.react.js:18]: @felipefialho Yes, is the same configuration that react-native https://github.com/facebook/react-native/blob/c267a0679d21908e6e8026ca5ee542a9b571fb8b/packages/eslint-config-react-native/index.js#L311

**Review by gabrielduete — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (felipefialho) [reply package.json:39]: Uhmmm... so, make sense to keep this dependency inside project?

**Review by felipefialho — APPROVED** [review]

**Pull request merged**